subsys/shell/Kconfig

```diff
@@ -38,6 +38,7 @@ menuconfig SHELL
 	bool "Enable shell"
 	select LOG_RUNTIME_FILTERING
 	select POLL
+	select SERIAL
```

Comment:
`depends on` might be better here. `select`ing symbols with prompts or dependencies often leads to two common issues:

 - `select` ignores dependencies. If someone adds (unsatisfied) dependencies to `SERIAL`, this will still force it on (but at least you get a warning).

 - With `select`, you can't immediately tell when looking at `SERIAL` that there's other ways for it to get enabled, or that `CONFIG_SHELL=y` also enables `SERIAL`. This can lead to twisty and hard-to-follow Kconfig files if there's a lot of `select`ing going on.

Usually, it's best to limit `select` to simple invisible "helper" symbols (`select CPU_SUPPORTS_FOO` and the like). It's great for that.

I whitelisted the select-without-satisfied-dependencies warning when I was turning warnings into errors, because there was a lot of it going on, and a lot of it seemed to be from `CONSOLE`-related symbols. Should probably refactor a bunch of those to use `depends on` instead. :)

Reply:
This sentiment also appears in `kconfig-language.txt` btw:

```
select should be used with care. select will force a symbol to a value without
visiting the dependencies. By abusing select you are able to select a symbol
FOO even if FOO depends on BAR that is not set. In general use select only for
non-visible symbols (no prompts anywhere) and for symbols with no dependencies.
That will limit the usefulness but on the other hand avoid the illegal
configurations all over.
```

Common sense overrides it though, but in practice, `select` very often leads to those issues.

Reply:
Agreed.